logging/internal/environment.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/compute/metadata"
+)
+
+// ResourceAtttributesGetter declares interface to lookup environment variables and metadata attributes
+type ResourceAtttributesGetter interface {
+	EnvVar(name string) (string, bool)
+	Metadata(path string) (string, bool)
+	ReadAll(path string) (string, error)
```

Comment:
Feels odd that this one returns an error and the others are bools. I wonder if you need a second return value at all. Could yoou cue off the string being "" to be a failed lookup? That should never be valid for you I think.